test/index.js

```diff
@@ -127,12 +127,32 @@ describe('Boom', () => {
 
         it('identifies a boom object', () => {
 
-            expect(new Boom.Boom('oops') instanceof Boom.Boom).to.be.true();
-            expect(Boom.badRequest('oops') instanceof Boom.Boom).to.be.true();
-            expect(new Error('oops') instanceof Boom.Boom).to.be.false();
-            expect(Boom.boomify(new Error('oops')) instanceof Boom.Boom).to.be.true();
-            expect({ isBoom: true } instanceof Boom.Boom).to.be.false();
-            expect(null instanceof Boom.Boom).to.be.false();
+            const BadaBoom = class extends Boom.Boom { };
+
+            expect(new Boom.Boom('oops')).to.be.instanceOf(Boom.Boom);
+            expect(new BadaBoom('oops')).to.be.instanceOf(Boom.Boom);
+            expect(Boom.badRequest('oops')).to.be.instanceOf(Boom.Boom);
+            expect(new Error('oops')).to.not.be.instanceOf(Boom.Boom);
+            expect(Boom.boomify(new Error('oops'))).to.be.instanceOf(Boom.Boom);
+            expect({ isBoom: true }).to.not.be.instanceOf(Boom.Boom);
+            expect(null).to.not.be.instanceOf(Boom.Boom);
+        });
+
+        it('returns false when called on sub-class', () => {
+
+            const BadaBoom = class extends Boom.Boom {};
+
+            expect(new Boom.Boom('oops')).to.not.be.instanceOf(BadaBoom);
+            expect(new BadaBoom('oops')).to.not.be.instanceOf(BadaBoom);
```

Comment:
I'm confused by this as I would expect it to be: `expect(new BadaBoom('oops')).to.be.instanceOf(BadaBoom);` but there might be something I'm missing.

```js
class A {
  static [Symbol.hasInstance](instance) {
    // I believe that's the path that is executed in Boom when using Badaboom
    // since this == Badaboom and instance == (new BadaBoom('oops'))
    return this.prototype.isPrototypeOf(instance);
  }
}

class B extends A {}

const b = new B();

console.log(b instanceof B); // true

```

Reply:
You forget that the Boom constructor does not actually return an instance of the class. Ie. it won't match a regular instanceof.

Reply:
Indeed I missed that part. Thank you @kanongil.